### DISCUSSION TITLE
Sharing `World`?

### DISCUSSION THREAD

**Discussion by ghost** [bevyengine/bevy#3507] [Q&A] [answered]
Some of my components keep references to entities by storing their `Entity` structs. Systems that process those components need to be able to load the referenced entities, but systems have access only to the input `Query` trait that can't do that. A `World` can, but is it idiomatic? Should I put my `World` behind a mutex and share it on a global level for all systems to access as needed?

**Comment by DJMcNab** [comment, marked as answer]
Why can they not use `Query::get`?

**Reply by ghost** [reply]
Sorry - missed that one. Still getting used to VSCode. Thank you.
